### PR TITLE
maybe fixes #71

### LINKED ISSUE #71: unale to run the example
Hi,

First i have installed lastest version off crossbar into a virtual pypy3 vritualenvironement.
Then I have installed wampy using pip inside the virtualenvironement.
Thn i have the following crossbar config [https://github.com/noisyboiler/wampy/blob/master/wampy/testing/configs/crossbar.json](url)
and the following example [https://github.com/noisyboiler/wampy/blob/master/docs/examples/services.py](url)
runing the router => crossbar start --config crossbar.json
And try to launch the service using the run command.
It fail

then i have cloned the repo and followed your instruction 
pip install --editable .[dev]
crossbar, and many other lib downgraded to very old version

```
  Found existing installation: autobahn 18.9.2
    Uninstalling autobahn-18.9.2:
      Successfully uninstalled autobahn-18.9.2
  Found existing installation: crossbar 18.9.2
    Uninstalling crossbar-18.9.2:
      Successfully uninstalled crossbar-18.9.2
  Found existing installation: wampy 0.9.17
    Uninstalling wampy-0.9.17:
      Successfully uninstalled wampy-0.9.17
  Running setup.py develop for wampy
Successfully installed autobahn-0.17.2 colorlog-3.1.4 coverage-4.5.1 crossbar-0.15.0 enum-compat-0.0.2 eventlet-0.20.1 flake8-3.5.0 gevent-websocket-0.10.1 mccabe-0.6.1 mock-1.3.0 pbr-4.3.0 py-1.7.0 pycodestyle-2.3.1 pyflakes-1.6.0 pytest-3.1.3 pytest-capturelog-0.7 shutilwhich-1.1.0 wampy
```
Running again the router & all the stuff as specified in the doc,
And again the same probleme

```
wampy run docs.examples.services:BinaryNumberService --config ./wampy/testing/configs/crossbar.json
Traceback (most recent call last):
  File "/home/erc/.virtualenvs/pypy3/bin/wampy", line 11, in <module>
    load_entry_point('wampy', 'console_scripts', 'wampy')()
  File "/home/rep/icham/golden/crossbar/wampy/wampy/cli/main.py", line 27, in main
    args.main(args)
  File "/home/rep/icham/golden/crossbar/wampy/wampy/cli/run.py", line 104, in main
    run(app, config_path)
  File "/home/rep/icham/golden/crossbar/wampy/wampy/cli/run.py", line 70, in run
    router = Crossbar(config_path)
  File "/home/rep/icham/golden/crossbar/wampy/wampy/peers/routers.py", line 43, in __init__
    with open(config_path) as data_file:
FileNotFoundError: [Errno 2] No such file or directory: './crossbar/config.json'
```

```
ls -al wampy/testing/configs/crossbar.json
-rw-rw-r-- 1 erc erc 1324 Oct 13 19:11 wampy/testing/configs/crossbar.json
```

### PR DESCRIPTION
**Pull request by noisyboiler**
it does

### CODE REVIEW

**Comment by keiser1080** [comment]
Thanks for your reactivity.
I will test this asap.

File: test/test_app_runner.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        return bin(number)
+
+
+class TestAppRunne(object):
+
+    @pytest.fixture
+    def config_path(self):
+        return './wampy/testing/configs/crossbar.json'
+
+    def test_app_runner(self, router, config_path):
```
Comment (noisyboiler) [inline test/test_app_runner.py]: this test will run forever as is. will follow this up later :)

**Comment by keiser1080** [comment]
Hi,
I have created a fresh Python 3.5.2 virtual environement 
Then i have insyalled the fixes-issue-71
And i tryied to run the fix using your example (like in the docs)
First i get an error due to the crossbar configure file location.
Then i manage this first error by copying the file under the default location "./crossbar/config.json"
then i get a second error.

 
```
(wampyfix) ThinkPad-X250:~mkvirtualenv wampyfix -p /usr/bin/python3
(wampyfix) ThinkPad-X250:~workon wampyfix
(wampyfix) ThinkPad-X250:~pip install -e git://github.com/noisyboiler/wampy.git@fixes-issue-71#egg=wampyfix

(wampyfix) ThinkPad-X250:~wampy run docs.examples.services:DateService --config ./wampy/testing/configs/crossbar.json
Traceback (most recent call last):
  File "/home/icham/.virtualenvs/wampyfix/bin/wampy", line 11, in <module>
    load_entry_point('wampy', 'console_scripts', 'wampy')()
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/cli/main.py", line 27, in main
    args.main(args)
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/cli/run.py", line 103, in main
    run(app, config_path)
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/cli/run.py", line 70, in run
    router = Crossbar(config_path)
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/peers/routers.py", line 43, in __init__
    with open(config_path) as data_file:
FileNotFoundError: [Errno 2] No such file or directory: './crossbar/config.json'


# workaround
(wampyfix) ThinkPad-X250:~/dev/wamp/wampy$ mkdir crossbar
(wampyfix) ThinkPad-X250:~/dev/wamp/wampy$ cp wampy/testing/configs/crossbar.json ./crossbar/config.json


(wampyfix) ThinkPad-X250:~/dev/wamp/wampy$ wampy run docs.examples.services:DateService --config ./wampy/testing/configs/crossbar.json
Traceback (most recent call last):
  File "/home/icham/.virtualenvs/wampyfix/bin/wampy", line 11, in <module>
    load_entry_point('wampy', 'console_scripts', 'wampy')()
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/cli/main.py", line 27, in main
    args.main(args)
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/cli/run.py", line 103, in main
    run(app, config_path)
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/cli/run.py", line 70, in run
    router = Crossbar(config_path)
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/peers/routers.py", line 69, in __init__
    self.parse_url()
  File "/home/icham/.virtualenvs/wampyfix/src/wampyfix/wampy/mixins.py", line 28, in parse_url
    scheme, url = self.url.split(":", 1)
ValueError: not enough values to unpack (expected 2, got 1)
```

**Comment by keiser1080** [comment]
Hi,

fnally i manage to get wampy working.
I think the problem is due to the evolution between each realaese.
The router class signature change, and it's  in conflict with the run & runner .

here a working client & runner:

```
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import datetime
from wampy.backends import async_adapter
from wampy.peers.clients import Client
from wampy.roles.callee import callee
from wampy.roles.subscriber import subscribe
from wampy.peers.routers import Crossbar


class AppRunner(object):

    def __init__(self):
        self.apps = []

    def add_app(self, app):
        self.apps.append(app)
        
    def run(self):
        for app in self.apps:
            print("starting up app: %s" % app.__class__.__name__)
            app.start()
            print("{} is now running and connected.".format(app.__class__.__name__))
        print('all services started!')
        try:
            while True:
                self.wait()
        except KeyboardInterrupt:
            print('stopping all!!')
            self.stop()
    def stop(self):
        for app in self.apps:
            print("stopping app: %s" % app.__class__.__name__)
            app.stop()
            
    def wait(self):
        async_adapter.sleep()


def run(apps, url):
    print("starting up services...")
    runner = AppRunner()
    for app in apps:
        app = app(url=url)
        runner.add_app(app)     
    runner.run()
    
class DateService(Client):
    """ A service that returns the current date
    """  
    @callee
    def get_todays_date(self):
        return datetime.date.today().isoformat()

class SubscribingService(Client):
    """ A service that prints out "foo" topic messages
    """  
    @subscribe(topic="foo")
    def foo_handler(self, **kwargs):
        print("foo message received: {}".format(kwargs))

class BinaryNumberService(Client):
    """ A service that returns the binnary form of a number
    """  
    @callee
    def get_binary_number(self, number):
        return bin(number)

def main():
    url = 'ws://localhost:18080'
    apps = [DateService, SubscribingService, BinaryNumberService]
    run(apps, url)
    
if __name__ == '__main__':
    main()

```

**Pull request merged**